### PR TITLE
SCT-1147  Add utility fields and methods to mongo controller 

### PR DESCRIPTION
**Pull request by MrCreosote**

### CODE REVIEW

File: src/jgikbase/test/idmapping/mongo_controller.py [hunk 1]
```diff
@@ -56,6 +58,17 @@ def __init__(self, mongoexe: Path, root_temp_dir: Path, use_wired_tiger: bool=Fa
...
         time.sleep(1)  # wait for server to start up
+        self.client = MongoClient('localhost', self.port)
+        # check that the server is up. See
+        # https://api.mongodb.com/python/3.7.0/api/pymongo/mongo_client.html
+        #    #pymongo.mongo_client.MongoClient
+        self.client.admin.command('ismaster')
+
+        # get some info about the db
+        self.db_version = self.client.server_info()['version']
+        self.index_version = 2 if (semver.compare(self.db_version, '3.4.0') >= 0) else 1
```
Comment (jayrbolton) [inline src/jgikbase/test/idmapping/mongo_controller.py]: So I'm inferring from this code that mongo has two different versions for indexing the database, v2 after 3.4.0 and v1 before. FYI this definitely took a bit for me to realize what is going on here, so might be worth commenting a description.

**Comment by jayrbolton** [comment]
I just had one comment on this -- the logic around the indexing version could use some description

**Comment by jayrbolton** [comment]
Otherwise :+1:

File: src/jgikbase/test/idmapping/mongo_controller.py (hunk 1 above)
Comment (MrCreosote) [reply src/jgikbase/test/idmapping/mongo_controller.py]: done

**Comment by MrCreosote** [comment]
@jayrbolton verbally approved merge

**Pull request merged**